File: apps/mobile/test/removal_alerts_backbone_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:guardian/services/backbones/removal_alerts_backbone.dart';

void main() {
  test('removal-alerts stays hidden until its acceptance gates pass', () {
    expect(RemovalAlertsBackbone.lifecycle, 'implementation_disabled');
    expect(RemovalAlertsBackbone.enabledByDefault, isFalse);
    expect(RemovalAlertsBackbone.minimumPlan, 'family');
    expect(RemovalAlertsBackbone.customerVisible, isFalse);
    expect(RemovalAlertsBackbone.customerBuildEnabled, isFalse);
    expect(RemovalAlertsBackbone.protocolCommands, isNotEmpty);
    expect(RemovalAlertsBackbone.frontendMilestones, isNotEmpty);
    expect(RemovalAlertsBackbone.acceptanceGates, isNotEmpty);
  });
}

File: gateway/test/removal-alerts-backbone.test.js
const test = require('node:test');
const assert = require('node:assert/strict');

const {
  SERVICE_CONTRACT,
} = require('../src/service-backbones/removal-alerts');

test('removal-alerts remains a disabled implemented service', () => {
  assert.equal(SERVICE_CONTRACT.lifecycle, 'implementation_disabled');
  assert.equal(SERVICE_CONTRACT.enabledByDefault, false);
  assert.equal(SERVICE_CONTRACT.minimumPlan, 'family');
  assert.equal(SERVICE_CONTRACT.customerVisible, false);
  assert.ok(SERVICE_CONTRACT.protocolCommands.length > 0);
  assert.ok(SERVICE_CONTRACT.backendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.frontendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.acceptanceGates.length > 0);
  assert.ok(Object.isFrozen(SERVICE_CONTRACT));
});
